lib/core/api.js

```diff
@@ -166,8 +166,21 @@ api.user.updateProfile = function(userId, updates, cb){
     if(err){
       return cb(err);
     }
+    var currentEmail = _.get(profile, 'emails[0]');
     var newProfile = actionUtils.mergeProfileUpdates(profile, updates);
-    return api.user.addProfile(userId, newProfile, cb);
+    var emails = _.get(updates, 'emails');
+    if (_.isArray(emails) && emails.length === 1 && emails[0] !== currentEmail){
```

Comment:
If we were using it in more than one place, I'd say it might be a candidate for moving to a `util` function, but as is, a comment might be more appropriate.